From 919fdd56e5b166c5cf11eedf29a49fa866579ad0 Mon Sep 17 00:00:00 2001
From: Jamin Lin <jamin_lin@aspeedtech.com>
Date: Thu, 24 Sep 2026 17:18:30 +0800
Subject: [PATCH v1 20/30] hw/arm/aspeed_ast1040: Add I3C as an unimplemented
 device

The AST1040 provides an I3C at 0x74c20000, but the controller is not
modeled yet.

Add the I3C register region as an unimplemented device to reflect the
AST1040 memory map.

Signed-off-by: Jamin Lin <jamin_lin@aspeedtech.com>
Change-Id: I2dcb02f394f5510840b5522bf8266a373f7d00b2
---
 include/hw/arm/aspeed_soc.h | 2 +-
 hw/arm/aspeed_ast1040.c     | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/hw/arm/aspeed_soc.h b/include/hw/arm/aspeed_soc.h
index 709a805111..2d88fda346 100644
--- a/include/hw/arm/aspeed_soc.h
+++ b/include/hw/arm/aspeed_soc.h
@@ -63,7 +63,7 @@
 #define ASPEED_INTC_NUM  2
 #define ASPEED_IOEXP_NUM 2
 #define ASPEED_SRAM_NUM 2
-#define ASPEED_UNIMP_NUM 10
+#define ASPEED_UNIMP_NUM 14
 
 typedef struct AspeedUnimpDevice {
     const char *qom_name;
diff --git a/hw/arm/aspeed_ast1040.c b/hw/arm/aspeed_ast1040.c
index a851329197..678731f673 100644
--- a/hw/arm/aspeed_ast1040.c
+++ b/hw/arm/aspeed_ast1040.c
@@ -97,6 +97,7 @@ static const AspeedUnimpDevice aspeed_soc_ast1040_unimp_devs[] = {
     { "spi1-filter", "aspeed.spi1-filter", ASPEED_DEV_SPI1_FILTER, 0x1000 },
     { "otp", "aspeed.otp", ASPEED_DEV_OTP, 0x800 },
     { "atc", "aspeed.atc", ASPEED_DEV_ATC, 0x100 },
+    { "i3c", "aspeed.i3c", ASPEED_DEV_I3C, 0x8000 },
 };
 QEMU_BUILD_BUG_ON(ARRAY_SIZE(aspeed_soc_ast1040_unimp_devs) > ASPEED_UNIMP_NUM);
 
-- 
2.53.0

